llvm/utils/TableGen/Basic/CMakeLists.txt

```diff
@@ -9,8 +9,15 @@ set(LLVM_LINK_COMPONENTS
   )
 
 add_llvm_library(LLVMTableGenBasic OBJECT EXCLUDE_FROM_ALL DISABLE_LLVM_LINK_LLVM_DYLIB
+  ARMTargetDefEmitter.cpp
+  Attributes.cpp
   CodeGenIntrinsics.cpp
+  DirectiveEmitter.cpp
+  IntrinsicEmitter.cpp
+  RISCVTargetDefEmitter.cpp
   SDNodeProperties.cpp
+  TableGen.cpp
```

Comment:
I suggest `TableGen.cpp` may be individual. IIRC, some cmake generators don't like `add_executable` w/o any source files.

Reply:
CMake has a problem to identify which linker to use (C or C++) when there is no .c/.cpp file in the list. A strategy use elsewhere also in LLVM adds an empty `dummy.cpp` into the list.

Keeping `TableGen.cpp` (or using the same `dummy.cpp`) in both executables would defeat the entire purpose of this patch: The same timestamp issue still occurs with even just a single file which causes all .td files to be processed again in a no-change build.

I instead applied the usual source structure of an executable: One file per executable which named after the executable name containing the (in this case trivial) `main` function. This should also clear up any confusion (including mine) of where each main function is.